index/index.go

```diff
@@ -96,6 +96,13 @@ type IndexReader interface {
 	Close() error
 }
 
+// IndexReaderAdv is an optional interface for advanced users
+// Hope to have a better name here...
+type IndexReaderAdv interface {
```

Comment:
agree, i'll try to propose some names, and might recommend we use 2 separate interfaces